docs/kb/intro-functions.md

```diff
@@ -8,25 +8,324 @@ cta:
  actions:
+    - text: Install the [Appliance with Knative](install-knative) to extend your SDDC with our [community-sourced functions](/examples-knative)
     - text: Install the [Appliance with OpenFaaS](install-openfaas) to extend your SDDC with our [community-sourced functions](/examples)
     - text: Learn more about the [Events in vCenter](vcenter-events) and the [Event Specification](eventspec) used to send the events to a Function
     - text: Find steps to deploy a function - [instructions](use-functions).
 ---
 
 # Functions
 
-## Getting Started
+The VMware Event Broker Appliance can be deployed using either Knative or OpenFaaS event processor to provides customers with a Function-as-a-Service (FaaS) platform.
 
-The VMware Event Broker Appliance deployed with OpenFaaS provides a Function-as-a-Service (FaaS) platform. Alex Ellis, the creator of OpenFaaS, and the community have put together comprehensive documentation and workshop materials to get you started with writing your first functions:
+## Table of Contents
+- [Knative](#knative)
+  - [Knative Naming and Version Control](#knative-naming-and-version-control)
+    - [Knative Service](#knative-service)
+    - [Knative Trigger](#knative-service)
+    - [Knative Combined Service and Trigger](#knative-combined-service-and-trigger)
+    - [Knative Secrets](#knative-secrets)
+    - [Knative Environment Variables](#knative-environment-variables)
+- [OpenFaaS](#openfaas)
+  - [OpenFaaS Naming and Version Control](#openfaas-naming-and-version-control)
+
+## Knative
+
+Users who directly want to jump into VMware vSphere-related function code might want to check out the examples we provide [here](/examples/knative).
+
+### Knative Naming and Version Control
+
+When it comes to authoring functions, it's important to understand the different components that make up a Knative function deployment. Let's take the following excerpt as an example:
+
+### Knative Service
+
+A Knative Service `kn-service.yaml` defines the container image that will be executed upon invocation.
+
+```
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+ name: kn-ps-echo
+ namespace: vmware-functions
+spec:
+ template:
+  spec:
+   containers:
+    - image: projects.registry.vmware.com/veba/kn-ps-echo:latest
+```
+
+`kn-ps-echo`: The name of the Knative Service.
+
+The value of this field:
+
+- must not conflict with an existing function
+- should not contain special characters, e.g. "$" or "/"
+- should represent the intent of the function, e.g. "tag" or "tagging"
+
+`vmware-functions`: The Kubernetes namespace to deploy all functions to. By default, this is `vmware-functions` which is automatically created as part of the VMware Event Broker Appliance setup.
+
+> **Note**: It is recommended to deploy all functions to the default `vmware-functions` namespace since the VMware Event Broker Appliance UI is automatically been configured to manage all functions and secrets within this namespace.
```

Comment:
Good note! Might want to add that the broker lives there and would otherwise break fns

Reply:
👍